classes/class-wc-connect-nux.php

```diff
@@ -177,15 +198,22 @@ public static function get_banner_type_to_display( $status = array() ) {
 				case self::JETPACK_ACTIVATED_NOT_CONNECTED:
 					return 'before_jetpack_connection';
 				case self::JETPACK_CONNECTED:
-					// Has the user just gone through our NUX connection flow?
-					if ( isset( $status['should_display_after_cxn_banner'] ) && $status['should_display_after_cxn_banner'] ) {
-						return 'after_jetpack_connection';
-					}
+				case self::JETPACK_DEV:
 					// Has the user already accepted our TOS? Then do nothing.
 					// Note: TOS is accepted during the after_connection banner
-					if ( isset( $status['tos_accepted'] ) && ! $status['tos_accepted'] ) {
+					if (
+						isset( $status['tos_accepted'] )
+						&& ! $status['tos_accepted']
+						&& $this->can_accept_tos()
```

Comment:
Not sure that we should make this function non-pure. Can we pass `$this->can_accept_tos()` in as a parameter in the array & add unit tests for it?